src/include/sof/list.h

```diff
@@ -72,6 +74,18 @@ static inline void list_item_append(struct list_item *item,
 	list->prev = item;
 }
 
+static inline void list_item_append_smp(struct list_item *item,
+	struct list_item *list)
+{
+	list_item_append(item, list);
+
+	dcache_writeback_region(list, sizeof(*list));
```

Comment:
I was considering list_item_append_invalidate_writeback(), but in the end I decided, that I only had 1 or maybe a couple of uses for that, and that it's actually dangerous, it can drop your recent changes. Let's not add it in this patch, later if it turns out we have sufficiently many uses for it, we can add it.

Reply:
ok for list_item_append_invalidate_writeback() to be dropped.